ntoskrnl/ex/sysinfo.c

```diff
@@ -2847,7 +2847,8 @@ NtQuerySystemInformation(
         /*
          * Check if the request is valid.
          */
-        if (SystemInformationClass >= MAX_SYSTEM_INFO_CLASS)
+        if (SystemInformationClass < MIN_SYSTEM_INFO_CLASS ||
+            SystemInformationClass >= MAX_SYSTEM_INFO_CLASS)
```

Comment:
Someone won't be happy due to the lack of extra parentheses around each of the smaller/greater-than comparisons... ^^